eth/abc.py

```diff
@@ -1542,17 +1542,27 @@ def get_log_entries(self) -> Tuple[Tuple[bytes, Tuple[int, ...], bytes], ...]:
     #
     # State Transition
     #
+    @classmethod
     @abstractmethod
-    def apply_message(self) -> 'ComputationAPI':
+    def apply_message(
+            cls,
+            state: 'StateAPI',
+            message: MessageAPI,
+            transaction_context: TransactionContextAPI) -> 'ComputationAPI':
         """
-        Execution of a VM message.
+        Execution of a VM message. Typically used for sub-calls.
```

Comment:
As I get to understand this code better, maybe it would be clearer to say that `apply_message` is where the VM-specific logic is handled, but it ultimately hands off to `apply_computation` to create and execute computation. I'll update the docstrings to say something like that before merging.